src/components/CreateRecipe.tsx

```diff
@@ -2,50 +2,90 @@ import {
   IconButton,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  Input,
+  Text,
   Textarea,
+  useDisclosure,
+  Image,
+  FormControl,
 } from "@chakra-ui/react";
 import { HamburgerIcon } from "@chakra-ui/icons";
 import EditableName from "./EditableName.tsx";
 import BreadcrumbNav from "./BreadcrumbNav.tsx";
 import Add from "./Add.tsx";
-import {
-  AutoComplete,
-  AutoCompleteInput,
-  AutoCompleteItem,
-  AutoCompleteList,
-} from "@choc-ui/chakra-autocomplete";
+import { ChangeEvent, useEffect, useRef, useState } from "react";
+import CreateIngredientModal from "./CreateIngredientModal.tsx";
 import Ingredient from "./Ingredient.tsx";
-import { useState } from "react";
-
-const initialIngredients = [
-  "cibule",
-  "kuřecí stehna",
-  "mleté hovězí",
-  "mleté vepřové",
-  "rýže",
-  "jasmínová rýže",
-];
+import { NewIngredient } from "./types.ts";
+import { useMyStore } from "./store.tsx";
+import { Controller, useForm } from "react-hook-form";
+import AddRecipeModal from "./AddRecipeModal.tsx";
+import { useNavigate } from "react-router-dom";
 
 const CreateRecipe = () => {
-  const [selectedIngredients, setSelectedIngredients] = useState<string[]>([]);
-  const [ingredients, setIngredients] = useState<string[]>(initialIngredients);
+  const navigate = useNavigate();
+  const {
+    control,
+    handleSubmit,
+    register,
+    reset,
+    formState: { errors, isSubmitting },
+  } = useForm();
 
-  const handleIngredientClick = (ingredient: string) => {
-    setSelectedIngredients([...selectedIngredients, ingredient]);
-    const updatedIngredients = ingredients.filter(
-      (item) => item !== ingredient,
-    );
-    setIngredients(updatedIngredients);
+  const [image, setImage] = useState("");
+  const [error, setError] = useState(false);
+  const [selectedIngredients, setSelectedIngredients] = useState<
+    NewIngredient[]
+  >([]);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const cancelRef = useRef<HTMLInputElement>(null);
+  const {
+    isOpen: isPopupOpen,
+    onOpen: onPopupOpen,
+    onClose: onPopupClose,
+  } = useDisclosure();
+  const focusRef = useRef<HTMLInputElement>(null);
+  const { addRecipe } = useMyStore();
+
+  useEffect(() => {
+    onPopupOpen();
+  }, []);
+
+  const onSubmit = (data: any) => {
+    if (selectedIngredients.length > 0) {
+      const updatedSelectedIngredients: string[] = selectedIngredients.map(
+        (ingredient) => ingredient.id,
+      );
+
+      const updatedData = {
+        ...data,
+        ingredients: updatedSelectedIngredients,
+        id: Date.now().toString(36),
+      };
+      addRecipe(updatedData);
+      navigate("/recepty");
+    } else if (selectedIngredients.length === 0) {
+      setError(true);
+    }
   };
 
-  console.log({ selectedIngredients });
+  useEffect(() => {
+    if (selectedIngredients.length > 0) {
+      setError(false);
+    }
+  }, [selectedIngredients]);
```

Comment:
Nechceme tu chybu nastavovat jen po tom, co clovek klikne na `onSubmit`?

Reply:
ja ji nastavuju v onSubmit, tady ji vypinam :-) aby kdyz mi tam zacne svitit po kliku na submit "vyberte ingredienci", tak aby to zhasnulo ve chvili, kdy pridam ingredienci